### ISSUE TITLE
Warning   ReconcileError  secrets "my-cluster-kubeconfig" not found

### ISSUE THREAD

**Issue by guettli** [syself/cluster-api-provider-hetzner#202]
/kind bug

**What steps did you take and what happened:**

I followed the docs, but applying the created yaml fails:

```
guettli@p15$ k get events -A --sort-by=.metadata.creationTimestamp 

default     8m14s       Warning   ReconcileError                machinehealthcheck/my-cluster-control-plane-unhealthy-5m   error creating client and cache for remote cluster: error fetching REST client config for remote cluster "default/my-cluster": failed to retrieve kubeconfig secret for Cluster default/my-cluster: secrets "my-cluster-kubeconfig" not found
default     8m14s       Warning   ReconcileError                machinehealthcheck/my-cluster-md-0-unhealthy-5m            error creating client and cache for remote cluster: error fetching REST client config for remote cluster "default/my-cluster": failed to retrieve kubeconfig secret for Cluster default/my-cluster: secrets "my-cluster-kubeconfig" not found
default     8m35s       Warning   ReconcileError                machinedeployment/my-cluster-md-0                          failed to retrieve HCloudMachineTemplate external object "default"/"my-cluster-md-0": hcloudmachinetemplates.infrastructure.cluster.x-k8s.io "my-cluster-md-0" not found
default     8m35s       Normal    SuccessfulCreate              machinedeployment/my-cluster-md-0                          Created MachineSet "my-cluster-md-0-7465476f6d"
default     8m4s        Normal    ChangeLoadBalancerAlgorithm   hetznercluster/my-cluster                                  Changed load balancer algorithm
default     8m4s        Normal    CreateLoadBalancer            hetznercluster/my-cluster                                  Created load balancer
default     8m          Normal    SuccessfulCreate              hcloudmachine/my-cluster-control-plane-72v5w               Created new server with id 20435964
default     7m56s       Warning   ReconcileError                machinehealthcheck/my-cluster-control-plane-unhealthy-5m   error creating client and cache for remote cluster: error creating dynamic rest mapper for remote cluster "default/my-cluster": Get "https://142.132.242.98:443/api?timeout=10s": dial tcp 142.132.242.98:443: connect: connection refused
default     7m45s       Normal    AddedAsTargetToLoadBalancer   hetznercluster/my-cluster                                  Added new server with id 20435964 to the loadbalancer 719385
default     6m52s       Warning   ReconcileError                machinehealthcheck/my-cluster-control-plane-unhealthy-5m   error creating client and cache for remote cluster: error creating dynamic rest mapper for remote cluster "default/my-cluster": context deadline exceeded
default     6m31s       Warning   ReconcileError                machinehealthcheck/my-cluster-md-0-unhealthy-5m            error creating client and cache for remote cluster: error creating dynamic rest mapper for remote cluster "default/my-cluster": context deadline exceeded
default     66s         Normal    DetectedUnhealthy             machine/my-cluster-control-plane-rb8qb                     Machine default/my-cluster-control-plane-unhealthy-5m/my-cluster-control-plane-rb8qb/ has unhealthy node
```
What could be the reason?

According to these docs, the secret is called `hetzner` https://github.com/syself/cluster-api-provider-hetzner/blob/main/docs/topics/preparation.md#create-a-secret-for-hcloud-only

I changed this, and used that:

```
kubectl create secret generic my-cluster-kubeconfig --from-literal=hcloud=$HCLOUD_TOKEN
kubectl patch secret my-cluster-kubeconfig -p '{"metadata":{"labels":{"clusterctl.cluster.x-k8s.io/move":""}}}'
```

Now I get this error:

```
default     29s         Warning   ReconcileError                machinehealthcheck/my-cluster-md-0-unhealthy-5m            error creating client and cache for remote cluster: error fetching REST client config for remote cluster "default/my-cluster": failed to retrieve kubeconfig secret for Cluster default/my-cluster: secrets "my-cluster-kubeconfig" not found
default     29s         Warning   ReconcileError                machinehealthcheck/my-cluster-control-plane-unhealthy-5m   error creating client and cache for remote cluster: error fetching REST client config for remote cluster "default/my-cluster": failed to retrieve kubeconfig secret for Cluster default/my-cluster: secrets "my-cluster-kubeconfig" not found
```

**Comment by ninavalls** [comment]
I see this here from the events: 
```
default     8m35s       Warning   ReconcileError                machinedeployment/my-cluster-md-0                          failed to retrieve HCloudMachineTemplate external object "default"/"my-cluster-md-0": hcloudmachinetemplates.infrastructure.cluster.x-k8s.io "my-cluster-md-0" not found
```

And it looks like the loadbalancer and first control-plane came up? So the secret with the hcloud token were found in the controller... Doesn't look like an issue with the secret...
Are you able to provider the caph logs?

**Comment by batistein** [comment]
@guettli did you also changed the HetznerCluster object? From what i see a cluster is created right? some logs and the modified cluster-template.yaml would be helpful to understand. If there is a real problem with the secrets this would be also intersting to cover in the e2e tests.

**Comment by guettli** [comment]
I think I found the error now.

In the Hetzner GUI I called the project "foo-project". After renaming it to "my-cluster" it worked.

And I think this error message always pops up during startup:

> secrets "my-cluster-kubeconfig" not found

AFAIK this secret gets created after you run `k apply -f my-cluster.yaml`. Since the processes run in parallel, the error message comes up, even if things are allright. It just takes some seconds until the secret "my-cluster-kubeconfig" gets created. At least that's what I think.

I am closing this one, since now my one-node cluster is running.

Thank you very much for your support.
